src/phoenix/server/query_handlers.py

```diff
@@ -0,0 +1,127 @@
+import asyncio
+from functools import partial
+from typing import AsyncIterator, Optional, cast
+
+import pandas as pd
+import pyarrow as pa
+from starlette.endpoints import HTTPEndpoint
+from starlette.requests import Request
+from starlette.responses import Response, StreamingResponse
+from starlette.status import HTTP_404_NOT_FOUND, HTTP_422_UNPROCESSABLE_ENTITY
+
+from phoenix.core.evals import Evals
+from phoenix.core.traces import Traces
+from phoenix.trace.dsl import SpanFilter, SpanQuery
+from phoenix.utilities import get_spans_dataframe, query_spans
+
+
+class GetSpansDataFrameHandler(HTTPEndpoint):
+    traces: Traces
+    evals: Optional[Evals] = None
+
+    async def post(self, request: Request) -> Response:
```

Comment:
Not blocking but POST is not the most standard for REST. Getting something is usually "GET" - I know we use graphql which breaks that convention but it might make sense to stick with conventions in this case

Reply:
Sounds good. In the interest of time, i'll get this deployed first. Then I'll enumerate all the refactoring tasks in a follow-up PR, because I need time to understand them better myself, but those changes don't impact the user.